src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,431 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.vector_measure
+import order.symm_diff
+
+/-!
+# Hahn decomposition
+
+This file prove the Hahn decomposition theorem (signed version).
+
+## Main results
+
+* `signed_measure.exists_disjoint_positive_negative_union_eq` : the Hahn decomposition theorem.
+* `signed_measure.exists_negative_set` : A measurable set of negative measure contains at least
+  one negative subset.
+
+## Notation
+
+We use the notations `0 ≤[i] v` and `v ≤[i] 0` to denote the usual definitions of a set `i`
+being positive/negative with respect to the signed measure `v`.
+
+## Tags
+
+Hahn decomposition theorem
+-/
+
+noncomputable theory
+open_locale classical big_operators nnreal ennreal
+
+variables {α β : Type*} [measurable_space α]
+variables {M : Type*} [add_comm_monoid M] [topological_space M] [ordered_add_comm_monoid M]
+
+namespace measure_theory
+
+namespace signed_measure
+
+open filter vector_measure
+
+variables {s : signed_measure α} {i j : set α}
+
+section exists_negative_set
+
+private def p (s : signed_measure α) (i j : set α) (n : ℕ) : Prop :=
+∃ (k : set α) (hj₁ : k ⊆ i \ j) (hj₂ : measurable_set k), (1 / (n + 1) : ℝ) < s k
```

Comment:
Is there a reason why you are using this formulation instead of the more standard
```
∃ k : set α, k ⊆ i \ j ∧ measurable_set k ∧ (1 / (n + 1) : ℝ) < s k
```
? Most of the time, we use existential quantifiers for data, and when there are only props left we use ... and ... and ...

Same thing several times below.

Reply:
I thought using that formulation is the professional thing to do :p I've changed it all to ...and... now